controller.go

```diff
@@ -198,7 +199,7 @@ func (c *controller) RegisterDriver(networkType string, driver driverapi.Driver,
 // NewNetwork creates a new network of the specified network type. The options
 // are network specific and modeled in a generic way.
 func (c *controller) NewNetwork(networkType, name string, options ...NetworkOption) (Network, error) {
-	if name == "" {
+	if name == "" || strings.Contains(name, ".") {
```

Comment:
I don't agree that because the cli decide to accept service and network in the form of  <service.network>,libnetwork backend has to enforce no dots in the  network name.